server/src/test/java/org/elasticsearch/cluster/action/shard/ShardStateActionTests.java

```diff
@@ -382,6 +382,33 @@ public void onFailure(Exception e) {
         assertThat(transport.capturedRequests(), arrayWithSize(0));
     }
 
+    public void testDeduplicateRemoteShardStarted() throws InterruptedException {
```

Comment:
Can we either add a test or randomly clear the deduplicator here and then validate we see two requests at the end?

Reply:
++ done